conf.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+)
+
+type JSONConf struct {
+	Branches struct {
+		Historical struct {
+			Development string `json:"development"`
+			//Poduction   string `json:"production"`
+		} `json:"historical"`
+		//Support struct {
+		//Feature string `json:"feature"`
+		//Release string `json:"release"`
+		//Hotfix  string `json:"hotfix"`
+		//Bugfix  string `json:"bugfix"`
+		//} `json:"support"`
+	} `json:"branches"`
+}
+
+func ReadConfiguration() JSONConf {
+	dir, _ := os.Getwd()
+
+	file, err := ioutil.ReadFile(dir + "/.git/conf.json")
+
+	c := JSONConf{}
+
+	// se non c'e' il file, ... allora imposto
+	// dei valori di default
+	if err != nil {
```

Comment:
That's not the only error, os.IsNotExist() can help :)

Reply:
OK but in this case we could have two use case: the file is readable or not. The latter includes both "no error" and "file not exists" case. This is the reason I prefer use nil. Am I wrong?

Reply:
Well IMO there are three case,
- no error (hence parse the contents)
- is not exist error (hence use defaults)
- other error (hence tell the user)

If the file becomes read-only or gets corrupted or the filesystem decides to fail the file read operation, I'd expect ff to bubble up the error and tell the user and exit, rather than use defaults.

Reply:
does `ioutil.ReadFile` can return `os.IsNotExist()`?